libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -567,10 +571,12 @@ void AP_GPS::detect_instance(uint8_t instance)
     // user has to explicitly set the MAV type, do not use AUTO
     // do not try to detect the MAV type, assume it's there
     case GPS_TYPE_MAV:
-#ifndef HAL_BUILD_AP_PERIPH
+#ifndef HAL_BUILD_AP_PERIPH 
+#if AP_GPS_MAV_ENABLED
```

Comment:
Need to be consistent around whether this goes around the `case` or inside the case.

My preference is outside the case and also `#if` in the header file in the enumeration - that way the compiler can help you eliminate code relating to the backend